mlflow/transformers/__init__.py

```diff
@@ -940,6 +1031,10 @@ def log_model(
         code_paths=code_paths,
         signature=signature,
         input_example=input_example,
+        # NB: We don't validate the serving input for Transformers flavor because
+        # it requires loading the model into memory and make prediction, which is
+        # expensive and can cause OOM errors.
+        validate_serving_input=False,
```

Comment:
Can we make this conditional? If a pipeline is specified, the object is already in memory and the benefits of input validation checking far outweigh the minor increase in runtime. If the model object reference is a uri path, then disable this option.

Reply:
Hmm I see. My concern was that we've already got multiple  complaints about running model prediction during logging. Now we run model three times total during logging (signature + dependency + serving input), which is costly and can cause OOM even for medium sized models.

We can make it conditional now, but should we also consider running them mode efficiently?

Reply:
I think a good follow-up is to refactor so that all 3 of these are calculated with a single prediction call, if possible. At worst, combining the signature and serving validation calls into a single evaluation call will be worthwhile. 
Having this additional stage turned off for the use case of extreme model sizes for now is a good interim solution until we have some time to rethink combining these expensive invocation stages